### PR TITLE
detect-detection: Convert unittests to new FAIL/PASS API

### PR DESCRIPTION
**Pull request by hsadia538**
- [X] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [X] I have signed the Open Information Security Foundation contribution agreement at https://suricata.io/about/contribution-agreement/
- [ ] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Previous PR: https://github.com/OISF/suricata/pull/8108

Link to [redmine](https://redmine.openinfosecfoundation.org/issues/4019) ticket:

Describe changes:

- Update detect-detection unit tests to FAIL/PASS APIs

### CODE REVIEW

File: src/detect-detection-filter.c [hunk 1]
```diff
@@ -304,12 +305,10 @@ static int DetectDetectionFilterTestParse02 (void)
 {
     DetectThresholdData *df = NULL;
     df = DetectDetectionFilterParse("track both,count 10,seconds 60");
-    if (df && (df->track == TRACK_DST || df->track == TRACK_SRC) && (df->count == 10) && (df->seconds == 60)) {
-        DetectDetectionFilterFree(NULL, df);
-        return 0;
-    }
+    FAIL_IF_NOT_NULL(df);
+    DetectDetectionFilterFree(NULL, df);
```
Comment (victorjulien) [inline src/detect-detection-filter.c:309]: we can only get here if `df` is NULL, so this free statement is redundant

File: src/detect-detection-filter.c [hunk 2]
```diff
@@ -341,12 +341,11 @@ static int DetectDetectionFilterTestParse04 (void)
...
     DetectThresholdData *df = NULL;
     df = DetectDetectionFilterParse("count 10, track by_dst, seconds 60, count 10");
-    if (df && (df->track == TRACK_DST) && (df->count == 10) && (df->seconds == 60)) {
-        DetectDetectionFilterFree(NULL, df);
-        return 0;
-    }
+    FAIL_IF_NOT_NULL(df);
 
-    return 1;
+    DetectDetectionFilterFree(NULL, df);
```
Comment (victorjulien) [inline src/detect-detection-filter.c:346]: same

File: src/detect-detection-filter.c [hunk 3]
```diff
@@ -611,24 +593,19 @@ static int DetectDetectionFilterTestSig3(void)
...
-    else {
-        if (alerts != 3)
-            printf("alerts: %d != 3: ", alerts);
-        if (drops != 3)
-            printf("drops: %d != 3: ", drops);
-    }
+    FAIL_IF_NOT(alerts == 3);
+    FAIL_IF_NOT(drops == 3);
 
     SigGroupCleanup(de_ctx);
```
Comment (victorjulien) [inline src/detect-detection-filter.c:599]: these can be removed

File: src/detect-detection-filter.c [hunk 4]
```diff
@@ -484,16 +473,15 @@ static int DetectDetectionFilterTestSig2(void)
...
+
+    FAIL_IF_NULL(de_ctx);
 
     de_ctx->flags |= DE_QUIET;
 
-    s = de_ctx->sig_list = SigInit(de_ctx,"alert tcp any any -> any 80 (msg:\"detection_filter Test 2\"; detection_filter: track by_dst, count 4, seconds 60; sid:10;)");
-    if (s == NULL) {
-        goto end;
-    }
+    s = DetectEngineAppendSig(de_ctx,
```
Comment (victorjulien) [inline src/detect-detection-filter.c:481]: declare `s` on first use

**Review by victorjulien — CHANGES_REQUESTED** [review]
for the tests that you update, can you declare vars on first use?

File: src/detect-detection-filter.c (hunk 1 above)
Comment (hsadia538) [reply src/detect-detection-filter.c:309]: Ok . Thank you for reviews!

**Comment by hsadia538** [comment]
> for the tests that you update, can you declare vars on first use?

You mean all kinds of variable e.g `Packet *p `,`DetectThresholdData *df `etc

**Comment by jufajardini** [comment]
> > for the tests that you update, can you declare vars on first use?
> 
> You mean all kinds of variable e.g `Packet *p `,`DetectThresholdData *df `etc

Yes! :)

**Comment by hsadia538** [comment]
Followed by #8115

**Pull request closed without merging**